.github/api_best_practices.md

```diff
@@ -0,0 +1,198 @@
+# ACS API Guidelines
+An overview of the set of standardized guidelines and practices to ensure that the APIs across different 
+projects and teams adhere to a common structure and design. This consistency simplifies the process of understanding 
+and maintaining APIs. The guidelines helps identify potential pitfalls and design choices that could lead to issues to 
+help developers avoid common mistakes and build more robust APIs. When in doubt, It is encouraged to seek resolution
+in slack channel [#forum-acs-api-design](https://redhat-internal.slack.com/archives/C05MMG2PP8A).
+
+## Table of Contents
+
+- [Naming Guidelines](#naming-guidelines)
+  - [Service Name](#service-name)
+  - [Method Name](#method-name)
+  - [Message Name](#message-name)
+
+## Naming Guidelines
+
+### Service Name
+Service name **must** be unique and should use a noun that generally refers to a resource or product component 
+e.g. `DeploymentService`, `ReportService`, `ComplianceService`. Intuitive and well-known short forms or 
+abbreviations may be used in some cases (and could even be preferable) for succinctness 
+e.g. `ReportConfigService`, `RbacService`. 
+All RPC methods grouped into a single service must generally pertain to the primary resource of the service.
+
+All service defined in versioned package `/proto/api/v2` **must not** import from 
+`/proto/storage` and `/proto/internalapi` packages.
+All new RPC methods defined in versioned package `/proto/api/v1` **must not** import from
+`/proto/storage` and `/proto/internalapi` packages.
```

Comment:
This paragraph feels to not belong to Service Naming. Rather, this talks about proto files layout in repo and import rules.
I suggest moving it to a separate section, even if it's just declared as a stub. We'll be able to rename and move when the text is in a separate section, but we'll likely forget to move it after a couple iterations if this text stays here.

Reply:
These are guidelines but these aren't **naming** guidelines.

Anyhow, I don't feel pedantic enough to fight for that so it can stay here.

Reply:
I feel it impacts everything below, so I will leave it as is for now.